Cython/Compiler/ExprNodes.py

```diff
@@ -13062,6 +13063,20 @@ def find_special_bool_compare_function(self, env, operand1, result_is_bool=False
                     self.special_bool_cmp_utility_code = UtilityCode.load_cached("StrEquals", "StringTools.c")
                     self.special_bool_cmp_function = "__Pyx_PyString_Equals"
                     return True
+                elif result_is_bool:
+                    from .Optimize import optimise_numeric_binop
+                    result = optimise_numeric_binop(
+                        "Eq" if self.operator == "==" else "Ne",
+                        self,
+                        PyrexTypes.c_bint_type,
+                        self.operand1,
+                        self.operand2
+                    )
+                    if result:
+                        self.special_bool_cmp_function = result[0]
+                        self.special_bool_cmp_utility_code = result[1]
+                        self.special_bool_extra_args = result[2]
```

Comment:
I find this slightly better.
```suggestion
                        (self.special_bool_cmp_function,
                         self.special_bool_cmp_utility_code,
                         self.special_bool_extra_args,
                         _) = result
```